### PR TITLE
[widget] change tool window appearance

### PR DESCRIPTION
**Pull request by azdrojowa123**
Change tool window appearance. 
There is still a safety feature to be made against pressing a tool window button prematurely - logic written on `bsp-disable` branch will allow to do this. (For now if user click tool window button too early (before `BspInitializer#runActivity` finish work) tool window will not render correctly). 
We haven't talked about updating tool window & indexing - before my implementation when unloaded target was loading tool window refresh noticeably (usually). Now it doesn't but it is easy to change

### CODE REVIEW

**Review by abrams27 — COMMENTED** [review]
random stuff

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt [hunk 1]
```diff
@@ -76,21 +92,22 @@ public class BspConnectionService(private val project: Project) {
...
     val head = errors.firstOrNull()
     head?.let {
-      errors.drop(1).forEach{ head.addSuppressed(it) }
+      errors.drop(1).forEach { head.addSuppressed(it) }
       throw head
     }
   }
 
   public companion object {
+    public var bspConnectionFile: LocatedBspConnectionDetails? = null
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt:110]: is it used?

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/MagicMetaModelService.kt [hunk 2]
```diff
@@ -19,6 +19,8 @@ public class MagicMetaModelService(private val project: Project) {
     magicMetaModel = MagicMetaModel.create(magicMetaModelProjectConfig, projectDetails)
   }
 
+  public fun isMagicMetaModelInitialized(): Boolean = this::magicMetaModel.isInitialized
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/services/MagicMetaModelService.kt:22]: is it used?

File: src/main/kotlin/org/jetbrains/plugins/bsp/ui/utils/Bsp.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.jetbrains.plugins.bsp.ui.utils
+
+import org.jetbrains.protocol.connection.LocatedBspConnectionDetails
+
+public object Bsp {
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/ui/utils/Bsp.kt:5]: like i said - we can move it the application service

**Review by abrams27 — COMMENTED** [review]
yea another random stuff

File: src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/DisconnectAction.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  override fun actionPerformed(e: AnActionEvent) {
+    val project = e.project
+    val bspConnectionService = project?.getService(BspConnectionService::class.java)
+    bspConnectionService?.disconnect()
+  }
+
+  public override fun update(e: AnActionEvent) {
+    val project = e.project
+    val connectionService = project?.getService(BspConnectionService::class.java)
+    e.presentation.isEnabled = connectionService?.isRunning() == true
```
[the remark this answers is not in the document]
Comment (azdrojowa123) [reply src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/DisconnectAction.kt:19]: from isRunning return Boolean?

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt (hunk 1 above)
Comment (azdrojowa123) [reply src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt:110]: yes it is used for reconnect action, but I need to turn it into map (probably)

File: src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/DisconnectAction.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (abrams27) [reply src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/DisconnectAction.kt:19]: ohh it's nullable, nvm, sorry xd

File: src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/ConnectAction.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  override fun actionPerformed(e: AnActionEvent) {
+    val project = e.project
+    val bspConnectionService = project?.getService(BspConnectionService::class.java)
+    bspConnectionService?.reconnect()
+  }
+
+  public override fun update(e: AnActionEvent) {
+    val project = e.project
+    val connectionService = project?.getService(BspConnectionService::class.java)
+    e.presentation.isEnabled = connectionService?.isRunning() == false
```
[the remark this answers is not in the document]
Comment (azdrojowa123) [reply src/main/kotlin/org/jetbrains/plugins/bsp/ui/actions/ConnectAction.kt:19]: in very early project opening stage e.project return null in update method so need to somehow prevent from NPE

File: src/main/kotlin/org/jetbrains/plugins/bsp/ui/utils/BspUtilService.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package org.jetbrains.plugins.bsp.ui.utils
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/ui/utils/BspUtilService.kt:1]: u can move it to `services` package

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt [hunk 7]
```diff
@@ -76,21 +94,22 @@ public class BspConnectionService(private val project: Project) {
...
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt:112]: is it used?

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt [hunk 8]
```diff
@@ -50,16 +53,31 @@ public class BspConnectionService(private val project: Project) {
...
   }
 
+  public fun reconnect(locationHash: String) {
+    val bspService = BspUtilService.getInstance()
+    bspService.connectionFile[locationHash]?.let {
+      connect(it)
+    }
+  }
+
+  public fun isRunning(): Boolean? {
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt:77]: u can remove `{`

File: src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt [hunk 9]
```diff
@@ -50,16 +52,30 @@ public class BspConnectionService(private val project: Project) {
     val launcher = createLauncher(bspIn, bspOut, client)
 
     val listening = launcher.startListening()
+    bspProcess = process
     server = launcher.remoteProxy
     client.onConnectWithServer(server)
     cancelActions = listOf(
+      Cancelable {
+        process.destroy()
+        process.waitFor()
```
[the remark this answers is not in the document]
Comment (abrams27) [reply src/main/kotlin/org/jetbrains/plugins/bsp/services/BspConnectionService.kt]: bump

**Review by abrams27 — APPROVED** [review]

**Pull request merged**